### ISSUE TITLE
MSS problem with PPPoE and IPv6

### ISSUE THREAD

**Issue by alex1702** [opnsense/core#4484]
[x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md

[x] I have searched the existing issues and I'm convinced that mine is new.

**Describe the bug**
I have a WAN connection via PPPoE. IPv4 was working all the time without any problems and I could connect in and out through my server. But with IPv6 I had strange problems. I was able to connect from my server `ping6 ipv6.google.com` without any problems, but connections to some hosts (e.g. `curl -6 https://packages.gitlab.com/install/repositories/gitlab/gitlab-ce/script.deb.sh` or `ping6 packages.gitlab.com`) did not work and went into a timeout. Also in the other direction via ssh to the server I had problems.
I found the following article ([Telekom VDSL MTU and MSS Clamping for IPv4 and IPv6](https://webcodr.io/2018/02/telekom-vdsl-mtu-und-mss-clamping-f%C3%BCr-ipv4-und-ipv6/) - there is also an English version below) and based on it I set the MSS of my PPPoE interface to 1472 (1432 + 40 , the "40" are automatically subtracted).
So my IPv6 is now working without problems.

Is it possible that the MSS is set correctly to -40 for IPv4, but also to -40 instead of -60 for IPv6 (as mentioned in the article on the website)? 

**Expected behavior**
Automatically set MSS for IPv6 with -60 instead of -40 and maybe an extra MSS field for Ipv6?

**Screenshots**

![Screenshot_2020-11-26  taldsl  Schnittstellen heimdall bitnetz de](https://user-images.githubusercontent.com/1862815/100388442-e1ccff00-302a-11eb-8786-34cfe7178e4e.png)

**Environment**

OPNsense 20.7.5 (amd64, OpenSSL).
APU4D4 4 i211AT LAN / AMD GX-412TC CPU
Network Intel® I210-AT

**Comment by AdSchellevis** [comment]
@alex1702 it does seem reasonable to adjust the mss for ipv6 indeed. https://github.com/opnsense/core/commit/d70a1aae03a716508aca9439a7db0f1abbe34957 should fix this.

**Comment by fichtner** [comment]
Isn’t 60 only valid if PPPoE IPv4 connectivity is used? IPv4 + additional IPv6 header on top?

**Comment by AdSchellevis** [comment]
@fichtner let's reopen this and see if we can validate the correct number. The way I interpreted the value was 20 bytes for the IP header(https://en.wikipedia.org/wiki/IPv4#Packet_structure) combined with 20 bytes for the TCP header (https://en.wikipedia.org/wiki/Transmission_Control_Protocol#:~:text=Specifies%20the%20size%20of%20the,of%20options%20in%20the%20header.) 

Since IPv4 header is of variable size and IPv6's header is fixed 40 (https://en.wikipedia.org/wiki/IPv6_packet), I assumed the 60 bytes was indeed the correct one for IPv6.

**Comment by alex1702** [comment]
> @fichtner let's reopen this and see if we can validate the correct number. The way I interpreted the value was 20 bytes for the IP header(https://en.wikipedia.org/wiki/IPv4#Packet_structure) combined with 20 bytes for the TCP header (https://en.wikipedia.org/wiki/Transmission_Control_Protocol#:~:text=Specifies%20the%20size%20of%20the,of%20options%20in%20the%20header.)
> 
> Since IPv4 header is of variable size and IPv6's header is fixed 40 (https://en.wikipedia.org/wiki/IPv6_packet), I assumed the 60 bytes was indeed the correct one for IPv6.

Exactly, I also read that a little later. But unfortunately I am not that deep into the TCP/IP stack.

**Comment by fichtner** [comment]
@alex1702 Can you share your advanced DHCPv6 settings to be sure?

**Comment by alex1702** [comment]
Is it ok in German or rather in English?

![Screenshot_2020-11-27  taldsl  Schnittstellen heimdall bitnetz de(2)](https://user-images.githubusercontent.com/1862815/100483767-502cc280-30fa-11eb-9202-9fac2d2a64d4.png)
![Screenshot_2020-11-27  taldsl  Schnittstellen heimdall bitnetz de](https://user-images.githubusercontent.com/1862815/100483769-50c55900-30fa-11eb-85f3-6ffdbe02393f.png)

**Comment by fichtner** [comment]
German works for me. 😊 Ok so this means 60 is correct when we force IPv6 over PPPoE link (only) same as how IPv4 works over PPPoE. If we don’t the MSS may be too small, but that’s only a soft issue.

**Comment by NKnusperer** [comment]
I hade the exact same problem as @alex1702 and setting the MSS to 1472 resolved it.
VDSL via PPPoE and ` Use IPv4 connectivity` for IPv6 enabled.
Stale IPv6 connections happen when the remote server is blocking ICMPv6.

For me it is not clear if setting the MSS manually should still be required in this scenario or if this was fixed already and tis a regression now. Can someone please clarify?

**Comment by alex1702** [comment]
> For me it is not clear if setting the MSS manually should still be required in this scenario or if this was fixed already and this is a regression now. Can someone please clarify?

I have the current OPNsense version 21.1.7 and was able to remove the manual MSS value again.

**Comment by NKnusperer** [comment]
That's strange, I'm on the same version and it didn't worked automatically.
Can I somehow get the value when it is set automatically?

**Comment by NKnusperer** [comment]
Presumable with the `22.7.4` release this issue popped up again. Set the MSS to 1472 for now.

**Comment by c-vo** [comment]
The German provider “Telekom” is experiencing a problem in connection with the “Magenta-TV” web client. If 1472 is entered in MSS on the WAN interface (PPPoE), the TV web client works. If the field is left blank, access to the web client (https://web.magentatv.de/) is not possible. Can this behavior be related to the issue described here? opnsense version 23.7.3, IPv6 enabled.

**Comment by NKnusperer** [comment]
@c-vo yes, see https://telekomhilft.telekom.de/t5/TV/Magenta-TV-Web-funktioniert-an-IPv6-Anschluss-nicht/m-p/5235310#M472246

**Comment by c-vo** [comment]
@NKnusperer okay, of course that explains a lot. Thank you very much!

**Comment by ProTofik** [comment]
Hi. I have exactly the same problem with OPNSense 25.7.3_7 when trying to access some IPv6 websites. I'm in the UK and my broadband provider is Zen Broadband. I have narrowed it down to websites that use Azure CND such as www.o2.co.uk.

Unless I set MSS on my WAN interface to 1492 or lower, those websites time out during TLS negotiation.

OPNSense web interface says that whatever value I enter for MSS, it will use that value minus 40 for IPv4 or minus 60 for IPv6, and that is exactly what I see in packet capture. But why isn't it doing it on its know since it knows it's a PPPoE connection? With everything left on default, it subtracts from 1500 instead.
